kube-core/src/version.rs

```diff
@@ -137,7 +207,7 @@ mod tests {
             Version::Stable(2),
             Version::Beta(2, Some(3)),
         ];
-        vers.sort();
+        vers.sort_by_cached_key(|x| Reverse(x.clone()));
```

Comment:
This is the only semi-annoying consequence of the reversal here.
A clone is forced on us from https://github.com/rust-lang/rust/issues/34162 unless we use the manual `cmp` calll reversed ourselves.

I think this is ok, the usage in kube_client does not need to clone because the Version parsing IS the temporary step, and the test here is probably less realistic of a use-case)

Reply:
`Ord` has a blanket impl that [`&T: Ord where T: Ord`](https://doc.rust-lang.org/stable/std/cmp/trait.Ord.html#impl-Ord-53), so I think this *should* line up correctly. Or do the lifetimes not match properly?

Reply:
End up getting this :/

```
error: lifetime may not live long enough
   --> kube-core/src/version.rs:208:37
    |
208 |         vers.sort_by_cached_key(|x| Reverse(x));
    |                                  -- ^^^^^^^^^^ returning this value requires that `'1` must outlive `'2`
    |                                  ||
    |                                  |return type of closure is Reverse<&'2 version::Version>
    |                                  has type `&'1 version::Version`
```